metrics.go

```
@@ -33,22 +41,58 @@ func registerMetrics() {
 	durationHistogram = promauto.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: "smtprelay",
 		Name:      "request_duration",
+		Help:      "duration of message relay requests",
 		Buckets:   prometheus.DefBuckets,
 	}, []string{"error_code"})
 
 	msgSizeHistogram = promauto.NewHistogram(prometheus.HistogramOpts{
 		Namespace: "smtprelay",
 		Name:      "message_bytes",
+		Help:      "size of messages",
 		Buckets:   []float64{0.05 * mb, 0.1 * mb, 0.25 * mb, 0.5 * mb, 1 * mb, 2 * mb, 5 * mb, 10 * mb, 20 * mb},
 	})
 }
 
-func handleMetrics() {
+func handleMetrics(ctx context.Context, addr string) (*instrumentationServer, error) {
+	// Setup listeners first, so we can fail early if the address is in use.
+	httpListener, err := net.Listen("tcp", addr)
```

What's the reason/benefit for failing early here?